### PR TITLE
Tests: Add backward compatibility test for cluster state with blocks

### PR DESCRIPTION
**Pull request by tlrx**
This commit adds a bwc test when the cluster state has blocks. It refers to https://github.com/elastic/elasticsearch/pull/9203#issuecomment-94755847.

### CODE REVIEW

**Comment by tlrx** [comment]
@javanna when you have time (no hurry), could you please have a look a this? Thanks :)

File: src/test/java/org/elasticsearch/bwcompat/ClusterStateBackwardsCompatTests.java [hunk 1]
```diff
@@ -52,4 +53,39 @@ public void testClusterState() throws Exception {
...
+        for (String blockSetting : Arrays.asList(SETTING_BLOCKS_READ, SETTING_BLOCKS_WRITE, SETTING_BLOCKS_METADATA)) {
+            try {
+                enableIndexBlock("test-blocks", blockSetting);
+
+                for (NodeInfo n : clusterNodes()) {
+                    try (TransportClient tc = newTransportClient()) {
+                        tc.addTransportAddress(n.getNode().address());
+
+                        ClusterStateResponse response = tc.admin().cluster().prepareState().setIndices("test-blocks")
+                                .setBlocks(false).setNodes(false).execute().actionGet();
```
Comment (javanna) [inline src/test/java/org/elasticsearch/bwcompat/ClusterStateBackwardsCompatTests.java]: why not ask for the blocks in the cluster state call?

**Comment by javanna** [comment]
left a small comment, looks good though, thanks a lot for taking care of this

File: src/test/java/org/elasticsearch/bwcompat/ClusterStateBackwardsCompatTests.java (hunk 1 above)
Comment (tlrx) [inline src/test/java/org/elasticsearch/bwcompat/ClusterStateBackwardsCompatTests.java]: I feel stupid now, I'm not testing everything here... I updated the code.

**Comment by tlrx** [comment]
@javanna updated, let me know what you think. Thanks.

**Comment by javanna** [comment]
LGTM

**Comment by tlrx** [comment]
Thanks @javanna

**Pull request merged**